lesson 6/Tensor/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,95 @@
+#include <jni.h>
+#include <string>
+#include <string.h>
+#include <sys/socket.h>	//socket
+#include <arpa/inet.h>	//inet_addr
+#include <unistd.h>
+#include <android/log.h>
+#include <errno.h>
+#include <locale>
+
+
+#define LOG_TAG "CLIENT_APP"
+
+jbyteArray getImage(jobject context, JNIEnv* env) {
+    jclass contextThemeWrapperClass = env->FindClass("android/view/ContextThemeWrapper");
+    jclass assetManagerClass = env->FindClass("android/content/res/AssetManager");
+    jclass inputStreamClass = env->FindClass("java/io/InputStream");
+
+    jmethodID getAssetsID = env->GetMethodID(contextThemeWrapperClass, "getAssets", "()Landroid/content/res/AssetManager;");
+    jmethodID openID = env->GetMethodID(assetManagerClass, "open", "(Ljava/lang/String;)Ljava/io/InputStream;");
+    jmethodID availableID = env->GetMethodID(inputStreamClass, "available", "()I");
+    jmethodID readID = env->GetMethodID(inputStreamClass, "read", "([B)I");
+    jmethodID closeID = env->GetMethodID(inputStreamClass, "close", "()V");
+
+    jobject assets = env->CallObjectMethod(context, getAssetsID);
+    jobject stream = env->CallObjectMethod(assets, openID, env->NewStringUTF("zivert.jpg"));
+
+    jint size = env->CallIntMethod(stream, availableID);
+    jbyteArray buffer = env->NewByteArray(size);
```

Comment:
Your implementation of this line was like this: 
`jbyteArray jbuffer[size];`
But the JNI arrays should be taken by `NewByteArray` method, so it should be:
`jbyteArray buffer = env->NewByteArray(size);`